### PR TITLE
@ashkan18 Fea opsworks stack

### PR DESCRIPTION
**Pull request by izakp**
Ok so this sets up an opsworks stack and provisioners for APR.  However, I can't for the life of me can't figure out this compile pipeline.

First of all, it expects prompts and there is no "--force" flag for mix so I have to pipe `yes` into the command (see below)

Second of all, I have to run "mix deps.get" and "mix compile" _as root_ before starting the phoenix server, and the at runtime (as the deploy user) it gives me:

```
=ERROR REPORT==== 31-May-2016::20:29:41 ===
file:path_eval([".","/root"],".erlang"): permission denied
Could not find Hex, which is needed to build dependency :phoenix
Shall I install Hex? [Yn]
```

@ashkan18 ,need some help on how to get this running under the `deploy` user...

### CODE REVIEW

File: chef/site-cookbooks/artsy_apr/recipes/deploy.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+application_env = case node['environment']
+when "production"
+  "prod"
+when "development"
+  "dev"
+end
+
+execute "get-mix-deps" do
+  command "MIX_ENV=#{application_env} yes | mix deps.get"
```
Comment (izakp) [inline chef/site-cookbooks/artsy_apr/recipes/deploy.rb]: @ashkan18 here is the invocation of deps.get

File: chef/site-cookbooks/artsy_apr/recipes/deploy.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  cwd "#{deploy_home}/current"
+  returns [0, 137]
+end
+
+execute "compile-mix-deps" do
+  command "MIX_ENV=#{application_env} yes | mix compile"
```
Comment (izakp) [inline chef/site-cookbooks/artsy_apr/recipes/deploy.rb]: and here is the invocation of compile

File: chef/site-cookbooks/artsy_apr/recipes/deploy.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  cwd "#{deploy_home}/current"
+  returns [0, 137]
+end
+
+execute "insall-npm-packages" do
+  command "npm install"
+  cwd "#{deploy_home}/current"
+end
+
+command = "mix phoenix.server"
```
Comment (izakp) [inline chef/site-cookbooks/artsy_apr/recipes/deploy.rb:74]: and here is the runtime command

**Comment by izakp** [comment]
@ashkan18 got it!  Dependencies installed and phoenix server runs.  Need to add the KAFKA url and we are good to go!

**Comment by ashkan18** [comment]
This is great! thanks @izakp 

Do we have to do anything specific to allow websockets?

**Comment by izakp** [comment]
Lol yes - fortunately we already tackled that with Causality... will update
the PR tomorrow

**Comment by ashkan18** [comment]
Awesome, I can also merge this one and have a separate PR for that update, whichever you prefer 👍 This is awesome! thanks!

**Comment by izakp** [comment]
Just leave this one open and we'll merge in one go :)

**Comment by izakp** [comment]
@ashkan18 okay ready to merge!

**Comment by ashkan18** [comment]
awesoooooooome!

**Pull request merged**